doc/source/contributing/index.rst

```diff
@@ -10,43 +10,126 @@ with this guide before attempting to contribute to PyConverter-XML2Py.
+    .. code:: bash
+
+        git clone https://github.com/ansys/pyconverter-xml2py
+
+#. Create a fresh-clean Python environment and activate it. If you require
+   additional information on creation of a virtual environment, see the
+   official Python `venv`_ documentation.
+
+    .. tab-set::
+
+      .. tab-item:: Linux
+        :sync: linux
+
+        ::
+
+          python -m venv venv
+          source venv/bin/activate
+
+      .. tab-item:: macOS
+        :sync: macos
+
+        ::
+
+          python -m venv venv
+          source venv/bin/activate
+
+      .. tab-item:: Windows
+        :sync: windows
+
+        ::
+
+          python -m venv venv
+          .\venv\Scripts\activate
+
+
+#. Make sure you have the latest version of `pip`_:
+
+    .. code:: bash
+
+        python -m pip install -U pip
+
+#. Install the project in editable mode:
+
+    .. code:: bash
+
+        python -m pip install -e .
+
+#. Install additional requirements (if needed):
+
+     .. code:: bash
+
+        python -m pip install .[doc,tests]
 
-    git clone https://github.com/ansys/pyconverter-xml2py
-    cd pyconverter-xml2py
-    python -m pip install --upgrade pip
-    pip install -e .
+#. Verify your development installation:
 
-Post issues
------------
+    .. code:: bash
 
-Use the `PyConverter-XML2Py Issues <pyconverter.xml2py_issues_>`_ page to submit questions,
-report bugs, and request new features. When possible, use these issue
-templates:
+        pytest tests -v
 
-* Bug report template
-* Feature request template
 
-If your issue does not fit into one of these categories, create your own issue.
+Style and testing
+-----------------
 
-To reach the PyAnsys support team, email `pyansys.core@ansys.com <pyansys.core@ansys.com>`_.
+If required, you can call style commands (such as `black`_, `isort`_,
+and `flake8`_) or unit testing commands (such as `pytest`_) from the command line.
+However, this does not guarantee that your project is being tested in an isolated
+environment, which is why you might consider using `tox`_.
 
 
+Documentation
+-------------
 
-Build documentation
--------------------
+For building documentation, you can run the usual rules provided in the
+`Sphinx`_ Makefile or make.bat:
 
-To build the PyConverter-XML2Py documentation locally, in the root directory of the repository,
-run these commands::
-    
-    pip install .[doc]
-    .\doc\make.bat html 
+.. tab-set::
+
+  .. tab-item:: Makefile
+
+    ::
+
+      python -m pip install .[doc]
+      make -C doc/ html
+
+
+  .. tab-item:: make.bat
+
+    ::
+
+      python -m pip install .[doc]
+      .\doc\make.bat html
+
+Once the documentation is built, you can open it as follow.
+
+.. tab-set::
+
+  .. tab-item:: Linux
+      :sync: linux
+
+      By running the command: 
+      ::
+
+        your_browser_name doc/html/index.html
+
+  .. tab-item:: macOS
+      :sync: macos
+
+      By opening the documentation from your file explorer located in ``doc/html/index.html``.
```

Comment:
```suggestion
      By opening the documentation. Use your file explorer to go to and open ``doc/html/index.html``.
```
Is file explorer here used generically--and not a reference to Windows File Explorer, which would have initial capital letters to reflect the proper noun?